### PR TITLE
Update FolderStructure.md

### PR DESCRIPTION
**Pull request by pscohn**
Closes https://github.com/rainforestapp/frontend-conventions/issues/21

### CODE REVIEW

File: FolderStructure.md [hunk 1]
```diff
@@ -4,38 +4,48 @@
...
-    action.js
-    reducer.js
+    SomeComponent.spec.js
+    SomeSubComponent.spec.js
+    SomeComponentActions.spec.js
+    SomeComponentReducer.spec.js
     mockData.js
   __mocks__
     index.js
+    SomeComponent.mock.js
```
Comment (AJFunk) [inline FolderStructure.md]: can/should we name mock files with `.mock.js`? We aren't doing this currently (same question for `.spec.js` in `__tests__`

File: FolderStructure.md [hunk 2]
```diff
@@ -4,38 +4,48 @@
...
-    index.js
-    action.js
-    reducer.js
+    SomeComponent.spec.js
+    SomeSubComponent.spec.js
+    SomeComponentActions.spec.js
+    SomeComponentReducer.spec.js
     mockData.js
   __mocks__
     index.js
```
Comment (AJFunk) [inline FolderStructure.md]: @pscohn after lots of debugging with @jkaipr, we discovered something that should be noted here.

If you are using `index.js` as the export location, the `__mocks__` directory needs to mimic the structure of the components (i.e. have a separate mock file for each component, import into `index.js` and export from there).

You need to import the actual component (not the mock) and it needs to have an absolute path. This allows it to be properly unmocked. See https://github.com/rainforestapp/regenwald/pull/3656/files#diff-057fdbe378b9542340e553f6e8f90926

File: FolderStructure.md (hunk 1 above)
Comment (jkaipr) [reply FolderStructure.md]: was wondering yesterday if this is configurable or not

Comment (jkaipr) [reply FolderStructure.md]: seems `.spec.js` and `.test.js` is supported, it's mentioned on the front page under `Zero configuration` https://facebook.github.io/jest/

Comment (jkaipr) [reply FolderStructure.md]: But reading a note in an [SO answer](https://stackoverflow.com/questions/40394013/naming-conventions-for-manual-mocks-in-jest-test-runner) seems like if you name your specs with these extensions, they are expected to live in the same directory as the components.

Comment (pscohn) [reply FolderStructure.md]: I think they're just saying it's a different option to solve whatever problem they're having, but using .spec.js inside of __tests__ works:

```
paul:regenwald $ npm test SetDisabled

> Regenwald@1.0.0 test /Users/paulcohn/rainforest/regenwald
> NODE_PATH=./node_modules:./src jest 1>/dev/null --colors "SetDisabled"

 PASS  src/app/v2/components/TestList/Controls/__tests__/SetDisabled.spec.js
```

Comment (jkaipr) [reply FolderStructure.md]: Ok. But seems like mock location / naming is currently not configurable https://github.com/facebook/jest/issues/2726, so we will have to have the same name for a component and it's mock.

Comment (pscohn) [reply FolderStructure.md]: That's unfortunate, I'd really like to have different names so hopefully they add that ability in the future

File: FolderStructure.md (hunk 2 above)
Comment (pscohn) [reply FolderStructure.md]: Added this, thanks!

**Comment by prayogoa** [comment]
We should figure out a plan to refactor the codebase to the new convention and make a jira ticket for it as part of merging this PR, discussion continued in the issue

**Comment by prayogoa** [comment]
Updated the PR to also include the new folder structure, this now closes #27

**Pull request merged**